chain/chain/src/chain.rs

```diff
@@ -585,6 +585,16 @@ impl Chain {
         })
     }
 
+    pub fn new_archival(
+        runtime_adapter: Arc<dyn RuntimeAdapter>,
+        chain_genesis: &ChainGenesis,
+        doomslug_threshold_mode: DoomslugThresholdMode,
+    ) -> Result<Chain, Error> {
+        let mut chain = Self::new(runtime_adapter, chain_genesis, doomslug_threshold_mode)?;
+        chain.store.set_archival();
+        Ok(chain)
+    }
```

Comment:
Yea, so in the commit message I mentioned that doing it this way makes it a way smaller change (30 line changes vs probably a couple hundred lines or something), and is easier to undo if we want to delete `ColTrieChanges` completely. 

So this kind of hinges on the other comment you have about the TODO comment. If we want to get rid of `ColTrieChanges` then maybe it makes sense to accept some ugliness here so that it will be easier to undo later. But if deleting `ColTrieChanges` is not on the horizon then I agree we should just change the params of `new()`. WDYT?

Reply:
No. Deleting the column and recomputing the data would slow down gc and we don’t want to do that. This is a space vs storage trade-off and we want want to pay the cost for increased speed. The comment you’ve linked was only in regards to reconstructing the data in case we need it for old state but as it turns out we never need it.

Reply:
@mina86 actually changed my mind on this. I think doing it like you suggested is better (adding an argument to `new()`). PTAL

in the end I don't think it's so bad/tedious after all